core/modules/field_ui/field_ui.admin.inc

```diff
@@ -57,28 +56,22 @@ function field_ui_fields_list() {
         if ($admin_path) {
           $bundle_manage_fields_link = $admin_path . '/fields';
           $field_configure_link = $bundle_manage_fields_link . '/' . $field_name;
-          $rows[$field_name]['data'][3][] = t('<a href="@field_configure_link">%field_label</a> in <a href="@bundle_manage_fields_link">@bundle_label</a>', array('@field_configure_link' => url($field_configure_link), '%field_label' => $field_label, '@bundle_manage_fields_link' => url($bundle_manage_fields_link), '@bundle_label' => $bundle_label));
+          $rows[$field_name]['data']['used_as'][] = t('<a href="@field_configure_link">%field_label</a> in <a href="@bundle_manage_fields_link">@bundle_label</a>', array('@field_configure_link' => url($field_configure_link), '%field_label' => $field_label, '@bundle_manage_fields_link' => url($bundle_manage_fields_link), '@bundle_label' => $bundle_label));
         }
         else {
-          $rows[$field_name]['data'][3][] = t('%field_label in @bundle_label', array('%field_label' => $field_label, '@bundle_label' => $bundle_label));
+          $rows[$field_name]['data']['used_as'][] = t('%field_label in @bundle_label', array('%field_label' => $field_label, '@bundle_label' => $bundle_label));
         }
+
       }
     }
   }
   foreach ($rows as $field_name => $cell) {
     // A reused field can have different labels across the various bundles it is
     // being used in. So we render the items of the "Used as" column as a list.
-    $rows[$field_name]['data'][3] = theme('item_list', array('items' => $rows[$field_name]['data'][3]));
+    $rows[$field_name]['data']['used_as'] = theme('item_list', array('items' => $rows[$field_name]['data']['used_as']));
   }
 
-  if (empty($rows)) {
-    $output = t('No fields have been defined yet.');
-  }
-  else {
-    // Sort rows by field name.
-    ksort($rows);
-    $output = theme('table', array('header' => $header, 'rows' => $rows, 'empty' => t('No fields available.'),));
-  }
+  $output = theme('table', array('header' => $header, 'rows' => $rows, 'empty' => t('No fields have been defined yet.'),));
```

Comment:
get sort+order with `tablesort_get_sort` & `tablesort_get_order` and the properly sort the `$rows` array before rendering as table.